angular/src/app/modules/review/review-detail/review-detail.component.ts

```diff
@@ -857,13 +866,172 @@ export class ReviewDetailComponent extends PagedListingComponentBase<ReviewDetai
   }
 
   sortByCountMonth(){
-    if(this.sortByMonth){
+    if(this.sortStatus){
       this.listReviewIntern = _.orderBy(this.listReviewIntern, ['countMonthLevelMax'], ['asc']);
     }
     else {
       this.listReviewIntern = _.orderBy(this.listReviewIntern, ['countMonthLevelMax'], ['desc']);
     }
-    this.sortByMonth = !this.sortByMonth;
+    this.sortStatus = !this.sortStatus;
+  }
+
+  sortByNewLevel(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['newLevel'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['newLevel'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+  
+  sortByLevel(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['currentLevel'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['currentLevel'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+
+  sortByUser(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['internName'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['internName'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+  
+  sortByReviewer(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['reviewerName'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['reviewerName'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+
+ 
+  sortByStatus(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['status'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['status'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+
+  setAll(completed: boolean) {
+    this.allComplete = completed;
+    if (this.listReviewIntern == null) {
+      return;
+    }
+
+    this.listSelectedItem = []; 
+
+    this.listReviewIntern.forEach((reviewDetail) => {
+      if (reviewDetail.status === 4) {
+        reviewDetail.selected = completed;
+
+        if (completed) {
+          this.listSelectedItem.push({
+            selected: completed,
+            id: reviewDetail.id,
+            status: reviewDetail.status,
+          });
+        }
+      }
+    });
+    this.checkSelectedCheckbox();
+  }
+
+  ishandleSelectlistReviewInternItem(index, $event) {
```

Comment:
Should change the function name to `handleSelectlistReviewInternItem`, not include `is`